[fennec] upgrade to GCC 6.1 or newer

The Fennec build is currently broken on Circle CI:

```
 0:02.06 checking the target C compiler version... 6.0.0
 0:02.24 checking the target C compiler works... yes
 0:02.24 checking for the target C++ compiler... /usr/bin/clang++-6.0
 0:02.32 checking whether the target C++ compiler can be used... yes
 0:02.32 checking the target C++ compiler version... 6.0.0
 0:02.39 checking the target C++ compiler works... yes
 0:02.39 checking for the host C compiler... /usr/bin/gcc
 0:02.42 checking whether the host C compiler can be used... no
 0:02.43 ERROR: Only GCC 6.1 or newer is supported (found version 5.4.0).
 0:02.45 *** Fix above errors and then restart with\
 0:02.45                "/usr/bin/make -f client.mk build"
 0:02.45 client.mk:145: recipe for target 'configure' failed
 0:02.45 make: *** [configure] Error 1
The command '/bin/sh -c ./mach build  && ./mach package' returned a non-zero code: 2
Exited with code 2
```

--- a/ubuntu-dev/ubuntu-dev.dockerfile
+++ b/ubuntu-dev/ubuntu-dev.dockerfile
@@ -78,10 +78,12 @@ RUN __LLVM_VERSION__="6.0" \
  && pip install --no-cache-dir --upgrade pip \
  && pip install --no-cache-dir --upgrade virtualenv \
  && pip install --no-cache-dir requests \
- && echo "SHELL=/bin/bash\nTERM=xterm-256color\nDISPLAY=:98\nCC=clang-${__LLVM_VERSION__}\nCXX=clang++-${__LLVM_VERSION__}" >> /etc/environment
+ && echo "SHELL=/bin/bash\nTERM=xterm-256color\nDISPLAY=:98\nCC=clang-${__LLVM_VERSION__}\nCXX=clang++-${__LLVM_VERSION__}\nHOST_CC=clang-${__LLVM_VERSION__}\nHOST_CXX=clang++-${__LLVM_VERSION__}" >> /etc/environment
 ENV SHELL /bin/bash
 ENV CC clang-6.0
 ENV CXX clang++-6.0
+ENV HOST_CC clang-6.0
+ENV HOST_CXX clang++-6.0
 
 # Disallow logging in to SSH with a password.
 RUN sed -ri "s/^[#\s]*PasswordAuthentication\s+[yn].*$/PasswordAuthentication no/" /etc/ssh/sshd_config \

